### PR TITLE
File-based registration

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
As detailed in the linked issue, this will take the output of the [previous pr](https://github.com/lyft/flytekit/pull/104) and register them with `flyte-cli`.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
This uses the same core logic as in the old registration path.  The only difference is that the data is read from files - files which follow a specific set of rules (detailed in help) - and the `to_flyte_idl()` call has already been made.

## Tracking Issue
https://github.com/lyft/flyte/issues/225

## Follow-up issue
NA

### CODE REVIEW

File: flytekit/clients/friendly.py [hunk 1]
```diff
@@ -5,16 +5,41 @@
...
 from flytekit.models import filters as _filters, common as _common, launch_plan as _launch_plan, task as _task, \
     execution as _execution, node_execution as _node_execution
 from flytekit.models.core import identifier as _identifier
 from flytekit.models.admin import workflow as _workflow, task_execution as _task_execution
+from flytekit.common.exceptions.user import FlyteAssertion as _FlyteAssertion
 
 
 class SynchronousFlyteClient(_RawSynchronousFlyteClient):
 
+    def register_entities(self, id_entity_list):
```
Comment (matthewphsmith) [inline flytekit/clients/friendly.py:20]: I think this is a little weird to have in the client because it is a non-transactional method and doesn't map directly to a single API call.

Comment (wild-endeavor) [reply flytekit/clients/friendly.py:20]: i can definitely move this back to the core cli code.

File: scripts/flytekit_build_image.sh [hunk 2]
```diff
@@ -61,8 +61,11 @@ if [ -n "$REGISTRY" ]; then
   if [[ "${REGISTRY}" == "docker.io"*  && -z "${NOPUSH}" ]]; then
     docker login --username="${DOCKERHUB_USERNAME}" --password="${DOCKERHUB_PASSWORD}"
   fi
-  docker push "${FLYTE_INTERNAL_IMAGE}"
-  echo "${FLYTE_INTERNAL_IMAGE} pushed to remote"
+
+  if [[ -z "${NOPUSH}" ]]; then
```
Comment (wild-endeavor) [inline scripts/flytekit_build_image.sh:65]: Clobbing this along this PR - just a minor bug fix that makes the building script respect the `NOPUSH` flag.

File: flytekit/clients/friendly.py [hunk 3]
```diff
@@ -51,6 +53,35 @@ def create_task(
             )
         )
 
+    def create_task_raw(
```
Comment (matthewphsmith) [inline flytekit/clients/friendly.py:56]: Why do we need to duplicate this? if you want to use this, why not use the raw client directly?
